GithubCommitProcessor/run.fsx

```diff
@@ -43,35 +43,46 @@ type PushWebHook = JsonProvider<pushSample>
 
 let Run(req: HttpRequestMessage, log: TraceWriter) =
     async { 
-        log.Info(sprintf 
-            "Handling a GitHub WebHook") 
+        log.Info "Handling a GitHub WebHook"
 
-        let dbContext = new GitHubData("fritzstreamdb")
+        // if dbContext is `IDisposable` you should use `use` here
+        // this will dispose `dbContext` at the end of this block (like `using` in C#)
+        // if it's not `IDisposable` change to
+        // let dbContext = GitHubData "fritzstreamdb"
+        // - because in F# the convention is to only use `new` if it's `IDisposable`
+        use dbContext = new GitHubData("fritzstreamdb")
 
         let! data = req.Content.ReadAsStringAsync() |> Async.AwaitTask
-        let webhook = PushWebHook.Parse(data)
+        let webhook = PushWebHook.Parse data
 
         let commitId = webhook.After
         let dateStamp = DateTime.UtcNow
         let repository = webhook.Repository.Name
-
-        let filteredCommits = webhook.Commits |> Seq.filter(fun c -> c.Committer.Username <> "web-flow") 
-
-        for commit in filteredCommits do
-          
-          let newRecord:Metric = {
-            Id = 0;
-            CommitId = commitId;
-            Repository = repository;
-            DateStamp = dateStamp;
-            Name = commit.Author.Username;
-            NumFilesChanged = commit.Added.Length + commit.Removed.Length + commit.Modified.Length;
-          }
-
-          dbContext.Metrics.Add newRecord |> ignore
+        
+        // code would be too ugly inline - but need the type-annotation here
+        // don't know exactly what the type-provider generates here
+        // but the trick is to run this, see the error and then
+        // input what the compiler tells you ;)
+        // (sorry for me being to lazy do download your files)
+        let addCommit (commit : PleaseInsertTheTypeHere_DontSeeIt) =
+          let newRecord = 
+            {
+                Id = 0
+                CommitId = commitId
+                Repository = repository
+                DateStamp = dateStamp
+                Name = commit.Author.Username
+                NumFilesChanged = commit.Added.Length + commit.Removed.Length + commit.Modified.Length
+            }
+          dbContext.Metrics.Add newRecord
+        
+        webhook.Commits 
+        |> Seq.filter (fun c -> c.Committer.Username <> "web-flow") 
+        |> Seq.iter addCommit
           
         let! loaded = dbContext.SaveChangesAsync() |> Async.AwaitTask 
-        loaded |> printfn "Loaded %d records" 
+        printfn "Loaded %d records" loaded
```

Comment:
IMO the pipe here makes it harder to read - usually I only use a single line `|>` if it reads naturally